src/test/regress/sql/partition_pruning.sql

```diff
@@ -973,4 +973,103 @@ where  trg.key1 = src.key1
 
 DROP TABLE t_part1;
 
+-- Tests of partition pruning for case when default partition was exchanged without validation
+-- and contains rows, that satisfy constraints of other partition
+
+-- simple delete test case
+create table test(i int, j int) partition by range(j) (start (1) end(3) every(2), default partition extra);
+insert into test values (0, 1);
+create table test_extra_exchanged(like test);
+insert into test_extra_exchanged values (0, 2);
+-- Perform exchange without validation
+set gp_enable_exchange_default_partition to on;
+alter table test exchange default partition with table test_extra_exchanged without validation;
+
+select tableoid::regclass, ctid, i, j from test;
```

Comment:
For current sql text and further please add comment lines about intention of sql blocks

Reply:
+